test/sqllogictest/uniqueness_propagation_filter.slt

```diff
@@ -322,8 +322,9 @@ query T multiline
 EXPLAIN WITH(arity, join implementations) SELECT DISTINCT f1, f2, f3, f4 FROM t2 WHERE f1 = f3 AND f2 = f4;
 ----
 Explained Query:
-  Filter (#0 = #2) AND (#1 = #3) // { arity: 4 }
-    ReadStorage materialize.public.t2 // { arity: 4 }
+  Project (#0, #1, #0, #1) // { arity: 4 }
```

Comment:
"regression", but .. hard to convince folks that among all projections, `#0 .. #3` would mean no projection.

Reply:
Actually, I guess not a regression but a strict improvement, because the MFP pushdown can explain that we don't actually need columns `#2` and `#3`. We don't print out MFP projections though, it seems?

Reply:
Yeah, I'd say this is fine. This will still be in the same MFP stage, which anyhow had an identity projection even in the old version, just not shown in MIR.

> We don't print out MFP projections though, it seems?

LIR always prints the projection of a `MapFilterProject`. However, in MIR an MFP is not represented as a `MapFilterProject`, but just as Map and Filter and Project. And if the Project doesn't need to do anything, then it disappears (e.g. during `CanonicalizeMfp`).

> Actually, I guess not a regression but a strict improvement, because the MFP pushdown can explain that we don't actually need columns #2 and #3

Yes.